apps/dashboard/app/apps/ood_app.rb

```diff
@@ -42,25 +42,42 @@ def title
   end
 
   def url
-    if manifest.url.empty?
+    if custom_url
+      self.class.fix_if_internal_url(custom_url, Rails.application.routes.url_helpers.root_path)
+    else
       if batch_connect_app?
         Rails.application.routes.url_helpers.new_batch_connect_session_context_path(token)
       else
         router.url
       end
-    else
-      custom_url = manifest.url % {
-        app_type: type,
-        app_owner: owner,
-        app_name: name,
-        app_token: token
-      }
-      self.class.fix_if_internal_url(custom_url, Rails.application.routes.url_helpers.root_path)
     end
   end
 
+  def custom_url
+    return nil unless manifest.url.present?
+
+    manifest.url % {
+      app_type: type,
+      app_owner: owner,
+      app_name: name,
+      app_token: token
+    }
+  end
+
+  def internal_url?
+    custom_url && self.class.relative_url?(custom_url)
+  end
```

Comment:
I feel like some test cases are in order. Shouldn't an app with no `custrom_url` (manifest.url) be internal here?

Reply:
An app with no custom url is always external

Reply:
I'll push that later with tests. Just keep this PR for the cosmetic fix.